sdk/storage/storage-queue/test/emulator-tests/miscellaneous.spec.ts

```diff
@@ -0,0 +1,54 @@
+import * as assert from "assert";
+import { MessagesClient, QueueClient, QueueServiceClient } from "../../src";
+import { getConnectionStringFromEnvironment, isBrowser } from "../utils";
+
+// Expected environment variables to run this test-suite
+// STORAGE_CONNECTION_STRING=UseDevelopmentStorage=true
+// TEST_MODE=emulator-tests
+describe("Emulator Tests", () => {
+  const messageContent = "Hello World";
+  let queueName: string;
+  let queueClient: QueueClient;
+  const env = isBrowser() ? (window as any).__env__ : process.env;
+  beforeEach(async function() {
+    if (env.TEST_MODE !== "emulator-tests") {
+      this.skip();
+    }
```

Comment:
The following three tests in this test suite will be skipped if the TEST_MODE is not "emulator-tests". 
As a result, they will be skipped during the record, playback and live test modes.

Post preview 3,  we can discuss about having the emulator for tests in the CI and potentially can all the tests against such emulator.

@jeremymeng,  @bterlson, @XiaoningLiu  Any thoughts?

Reply:
Could existing live tests run against emulator? Since it's only a different connection string in the env variable.  If that's the case I don't feel that we need special emulator tests.

Reply:
After you mentioned, tried to run all the tests with this amd related account name and key.
```
DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;
AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;
BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;
TableEndpoint=http://127.0.0.1:10002/devstoreaccount1;
QueueEndpoint=http://127.0.0.1:10001/devstoreaccount1;
```

Something weird happened.
```javascript
 const queueServiceClient = QueueServiceClient.fromConnectionString(
 getConnectionStringFromEnvironment()
 );
 
vs
 
 const queueServiceClient = getQSU();
```
Creating the `queueServiceClient ` from `getConnectionStringFromEnvironment` succeeded while from the `getQSU()` failed the tests.
Investigating.

Reply:
We can discuss about that.

Reply:
> Creating the queueServiceClient from getConnectionStringFromEnvironment succeeded while from the getQSU() failed the tests.

Reason behind this - 
 Endpoint for the emulator deviates from the syntax of the endpoint in `getQSU()`.
`https://${accountName}.queue.core.windows.net/`; 
whereas the emulator endpoint is - `http://127.0.0.1:10001/devstoreaccount1`

Fixed it by tweaking `getQSU()` to use the emulator connection string.

Reply:
Resolving this since all the questions were answered.